src/service-object.ts

```diff
@@ -366,10 +366,10 @@ class ServiceObject extends EventEmitter {
           this.create(...args);
           return;
         }
-        callback!(err, null, metadata as r.Response);
+        callback!(err, null, (metadata as unknown) as r.Response);
```

Comment:
The type for the third argument is wrong. It's suppose to be a `r.Response` but we're returning the metadata.

Reply:
I think this was totally wrong.  The last argument shouldn't be of type `r.Response`, it should be `Metadata`.  This was busted to begin with.

Reply:
In all other methods, the return type is `r.Response`.

Reply:
I don't think we've ever actually returned the entire response object though, just the response body?

Reply:
Exactly.  The original types were a mistake all over.

Reply:
The third argument should be a `ParsedHttpRespMessage`:

```ts
  /**
   * Sniff an incoming HTTP response message for errors.
   *
   * @param {object} httpRespMessage - An incoming HTTP response message from `request`.
   * @return {object} parsedHttpRespMessage - The parsed response.
   * @param {?error} parsedHttpRespMessage.err - An error detected.
   * @param {object} parsedHttpRespMessage.resp - The original response object.
   */
  parseHttpRespMessage(httpRespMessage: r.Response) {
    const parsedHttpRespMessage = {
      resp: httpRespMessage,
    } as ParsedHttpRespMessage;

    if (httpRespMessage.statusCode < 200 || httpRespMessage.statusCode > 299) {
      // Unknown error. Format according to ApiError standard.
      parsedHttpRespMessage.err = new ApiError({
        errors: new Array<GoogleInnerError>(),
        code: httpRespMessage.statusCode,
        message: httpRespMessage.statusMessage,
        response: httpRespMessage,
      });
    }

    return parsedHttpRespMessage;
  }
```

```ts
export interface ParsedHttpRespMessage {
  resp: r.Response;
  err?: ApiError;
}
```

It's not the response body either.

Reply:
Well this is funky. I knew that calling `Service#request` would yield the actual response as a third parameter, I didn't think we returned it anywhere. Taking a closer look it looks like that is not true for `getMetadata`, `setMetadata` and `delete` 😱, however in this case `get` does in fact return the response body as the third parameter.

Reply:
Heh, gross